### PR TITLE
Chore/replace activityobjects package with fork

### PR DESCRIPTION
**Pull request by TiMESPLiNTER**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-VUE_APP_API_CLIENT='server'
\ No newline at end of file
+VUE_APP_API_CLIENT="server"
+VUE_APP_BACKEND_URL="http://localhost:1337"
```
Comment (TiMESPLiNTER) [inline .env]: do we really want to check this in?

File: docker-compose.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
       context: .
       dockerfile: docker/dev/Dockerfile
     environment:
-      BACKEND_URL: https://mitra-social.herokuapp.com
+      VUE_APP_BACKEND_URL: https://mitra-social-backend.herokuapp.com
```
Comment (TiMESPLiNTER) [inline docker-compose.yml]: Guess this should stay `http://mitra-backend`, right?

File: src/api-client/mock/index.ts [hunk 3]
```diff
@@ -51,11 +57,34 @@ export default {
     }
     return await fetch(USER_TOKEN);
   },
+  async createUser(user: CreateUser) {
+    console.info(
+      `name: ${user.username}, email: ${user.email}, pwd: ${user.password}`
+    );
+    if (user.username === USER_NAME) {
+      return await error("User exists!");
```
Comment (TiMESPLiNTER) [inline src/api-client/mock/index.ts:65]:
```suggestion
      return await error("User already exists.");
```

File: src/api-client/mock/index.ts [hunk 4]
```diff
@@ -51,11 +57,34 @@ export default {
...
+    }
+
+    if (user.email === USER_EMAIL) {
+      return await error("Email exists!");
```
Comment (TiMESPLiNTER) [inline src/api-client/mock/index.ts]:
```suggestion
      return await error("This e-mail is already linked to an user.");
```

File: src/components/following/FollowingActor.vue [hunk 5]
```diff
@@ -25,9 +25,31 @@ export default class FollowingActor extends Vue {
 
   get icon(): string | undefined {
     const icon = ActivityObjectHelper.extractIcon(this.actor as ActivityObject);
```
Comment (TiMESPLiNTER) [inline src/components/following/FollowingActor.vue]:
```suggestion
    return ActivityObjectHelper.extractIcon(this.actor as ActivityObject);
```

File: src/store/modules/following.ts [hunk 6]
```diff
@@ -44,48 +47,77 @@ class Following extends VuexModule {
...
+      .then((actors) => {
+        this.context.commit("setFollowing", actors);
+      });
   }
 
   @Action
-  public async follow(payload: FollowPayload): Promise<void> {
-    const { to, object } = payload;
+  public async follow(actor: Actor): Promise<void> {
+    const oFollow = ActivityObjectHelper.normalizedObjectFollow(actor);
```
Comment (TiMESPLiNTER) [inline src/store/modules/following.ts]:
```suggestion
    const objectFollow = ActivityObjectHelper.normalizedObjectFollow(actor);
```

File: src/utils/activity-object-helper.ts [hunk 7]
```diff
@@ -14,25 +18,30 @@ export class ActivityObjectHelper {
...
-    }
-
-    if ((object as Actor).preferredUsername) {
+    } else if (
+      ActivityObjectHelper.hasProperty(object, "name") &&
+      (object as Actor).name
+    ) {
+      return (object as Actor).name;
+    } else if (
+      ActivityObjectHelper.hasProperty(object, "preferredUsername") &&
```
Comment (TiMESPLiNTER) [inline src/utils/activity-object-helper.ts]: I removed this `hasProperty` thing as I somewhere read it's not necessary and the second condition is enough `(object as Actor).preferredUsername` to check if the property is present.

File: src/views/Login.vue [hunk 8]
```diff
@@ -21,6 +21,7 @@
               prepend-icon="mdi-account"
               type="text"
               v-model="user"
+              v-on:keyup.enter="handleSubmit"
```
Comment (TiMESPLiNTER) [inline src/views/Login.vue]: 👍  could we maybe set this on the `v-form` element so we don't have to set it for every input field again?

File: src/views/SignUp.vue [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+      <v-card class="elevation-12">
+        <v-toolbar
+          :light="$vuetify.theme.dark"
+          :dark="!$vuetify.theme.dark"
+          flat
+        >
+          <v-toolbar-title>Sign Up</v-toolbar-title>
+        </v-toolbar>
+        <v-card-text>
+          <v-form lazy-validation v-model="valid" ref="signUpForm">
```
Comment (TiMESPLiNTER) [inline src/views/SignUp.vue]: Would be so cool if this works:

```suggestion
          <v-form lazy-validation v-model="valid" ref="signUpForm" v-on:keyup.enter="handleSubmit">
```

File: src/views/home/post/Posts.vue [hunk 10]
```diff
@@ -1,44 +1,54 @@
...
-          </v-list-item-content>
-        </v-list-item>
-        <v-divider class="mx-4"></v-divider>
+        <v-card-title>
+          <v-icon class="search-user-icon">mdi-account-search-outline</v-icon>
+        </v-card-title>
         <v-card-text>
-          <component :is="getComponent(post.type)" :data="post" />
+          You haven't got any posts yet because you're not following anyone yet.
+          Look for someone you can follow and enjoy reading
```
Comment (TiMESPLiNTER) [inline src/views/home/post/Posts.vue]:
```suggestion
          Look for someone you can follow and enjoy reading.
```

File: static.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
+  "root": "dist",
+  "clean_urls": true,
+  "routes": {
+    "/**": "index.html"
+  },
+  "proxies": {
+    "/api/": {
+      "origin": "https://mitra-social-backend.herokuapp.com"
```
Comment (TiMESPLiNTER) [inline static.json]: I don't think in general if we have any hardcoded URLs in the code... maye we can use an env variable here somehow as well?

File: tests/unit/components/actor/ActorPin.spec.ts [hunk 12]
```diff
@@ -25,8 +25,8 @@ describe("ActorPin.vue", () => {
       localVue,
       vuetify,
       propsData: {
-        attributedTo: (articles[0] as Article).attributedTo
-      }
+        actor: (articles[0] as Activity).actor,
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorPin.spec.ts]:
```suggestion
        actor: (activities[0] as Activity).actor,
```

File: tests/unit/components/actor/ActorPin.spec.ts [hunk 13]
```diff
@@ -71,24 +71,24 @@ describe("ActorPin.vue", () => {
...
+        actor: (articles[1] as Activity).actor,
+      },
     });
 
     expect(wrapper.find(".v-icon").exists()).toBe(true);
     expect(wrapper.find(".v-image").exists()).toBe(false);
   });
 
-  it("attributedTo is a string", () => {
+  it("attributedTo is a url", () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorPin.spec.ts]:
```suggestion
  it("actor is an url", () => {
```

File: tests/unit/components/actor/ActorSummarized.spec.ts [hunk 14]
```diff
@@ -28,54 +28,24 @@ describe("ActorSummarized.vue", () => {
...
-    };
-
-    const content = wrapper.find(".v-list-item__title");
-    expect(content.text()).toBe(names[lang]);
+  afterEach(() => {
+    AuthenticationUtil.clear();
   });
 
-  it("attributedTo is a string", () => {
+  it("attributedTo is a object with name property", () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorSummarized.spec.ts]:
```suggestion
  it("actor is an object with name property", () => {
```

File: tests/unit/components/actor/ActorSummarized.spec.ts [hunk 15]
```diff
@@ -28,54 +28,24 @@ describe("ActorSummarized.vue", () => {
...
+        actor: (articles[0] as Activity).actor,
+      },
     });
-
     const content = wrapper.find(".v-list-item__title");
-    expect(content.text()).toBe("johnny");
+    expect(content.text()).toBe("Sally");
   });
 
   it("attributedTo has a icon property as a image", () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorSummarized.spec.ts]:
```suggestion
  it("actor has an icon property as an image", () => {
```

File: tests/unit/components/actor/ActorSummarized.spec.ts [hunk 16]
```diff
@@ -175,87 +125,103 @@ describe("ActorSummarized.vue", () => {
...
 
+    await flushPromises();
     const followingRemoveIcon = wrapper.find(".mdi-account-remove");
+    const followingAddIcon = wrapper.find(".mdi-account-plus");
+    expect(followingAddIcon.exists()).toBe(false);
     expect(followingRemoveIcon.exists()).toBe(true);
   });
 
-  it("Actor is no following", () => {
+  it("Actor is no following", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorSummarized.spec.ts]:
```suggestion
  it("Actor is not followed by user", async () => {
```

File: tests/unit/components/actor/ActorSummarized.spec.ts [hunk 17]
```diff
@@ -175,87 +125,103 @@ describe("ActorSummarized.vue", () => {
...
 
+    await flushPromises();
+    const followingRemoveIcon = wrapper.find(".mdi-account-remove");
     const followingAddIcon = wrapper.find(".mdi-account-plus");
     expect(followingAddIcon.exists()).toBe(true);
+    expect(followingRemoveIcon.exists()).toBe(false);
   });
 
-  it("Follow an unfollow actor", async done => {
+  it("Following an unfollowed actor", async (done) => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorSummarized.spec.ts]:
```suggestion
  it("User follows an unfollowed actor", async (done) => {
```

File: tests/unit/components/actor/ActorSummarized.spec.ts [hunk 18]
```diff
@@ -175,87 +125,103 @@ describe("ActorSummarized.vue", () => {
...
-      expect(followingAddIcon.exists()).toBe(true);
+      followingAddIcon = wrapper.find(".mdi-account-plus");
+      expect(followingAddIcon.exists()).toBe(false);
       expect(followingRemoveIcon.exists()).toBe(true);
       done();
     });
   });
 
-  it("Unfollow an follow actor", async () => {
+  it("Unfollowing an followed actor", () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/actor/ActorSummarized.spec.ts]:
```suggestion
  it("User unfollows a followed actor", () => {
```

File: tests/unit/components/layouts/MitraHeader.spec.ts [hunk 19]
```diff
@@ -34,23 +34,20 @@ describe("Header.vue", () => {
...
+  it("User is set and has name 'john.doe'", async (done) => {
     const wrapper = mount(Header, { localVue, vuetify, store });
     store.dispatch("User/fetchUser", AuthenticationUtil.getUser()).then(() => {
       expect(wrapper.find(".user-content").text()).toBe("john.doe");
       done();
     });
   });
 
-  it("User is set and has avater", async done => {
+  it("User is set and has avater", async (done) => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/layouts/MitraHeader.spec.ts]:
```suggestion
  it("User is set and has an avater", async (done) => {
```

File: tests/unit/components/layouts/MitraHeader.spec.ts [hunk 20]
```diff
@@ -34,23 +34,20 @@ describe("Header.vue", () => {
     );
   });
 
-  it("User is set and has name 'john.doe'", async done => {
+  it("User is set and has name 'john.doe'", async (done) => {
```
Comment (TiMESPLiNTER) [inline tests/unit/components/layouts/MitraHeader.spec.ts]:
```suggestion
  it("User is set and has a name", async (done) => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 21]
```diff
@@ -0,0 +1,195 @@
...
+  };
+
+  beforeEach(() => {
+    vuetify = new Vuetify();
+    if (router.currentRoute.path !== "/signup") {
+      router.push({ name: "signup" });
+    }
+  });
+
+  it("SignUp success", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("User sign up is successful", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 22]
```diff
@@ -0,0 +1,195 @@
...
+
+    await flushPromises();
+    const button = wrapper.find("#submit");
+    button.trigger("click");
+
+    await flushPromises();
+    expect(router.currentRoute.path).toBe("/login");
+  });
+
+  it("User is empty", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Value for field `username` is empty", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 23]
```diff
@@ -0,0 +1,195 @@
...
+    wrapper.setData(data);
+
+    await flushPromises();
+    input.setValue("");
+
+    await flushPromises();
+    expect(wrapper.find(".v-messages__message").text()).toBe("Required.");
+  });
+
+  it("User is too short", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Value for field `username` is too short", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 24]
```diff
@@ -0,0 +1,195 @@
...
+    wrapper.setData(data);
+
+    await flushPromises();
+    input.setValue("");
+
+    await flushPromises();
+    expect(wrapper.find(".v-messages__message").text()).toBe("Required.");
+  });
+
+  it("Email is not valid 1", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Value for field `email` is not valid 1", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 25]
```diff
@@ -0,0 +1,195 @@
...
+    await flushPromises();
+    input.setValue("johnny.test");
+
+    await flushPromises();
+    expect(wrapper.find(".v-messages__message").text()).toBe(
+      "E-mail must be valid."
+    );
+  });
+
+  it("Email is not valid 2", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Value for field `email` is not valid 2", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 26]
```diff
@@ -0,0 +1,195 @@
...
+    await flushPromises();
+    input.setValue("johnny.test@mail");
+
+    await flushPromises();
+    expect(wrapper.find(".v-messages__message").text()).toBe(
+      "E-mail must be valid."
+    );
+  });
+
+  it("Email exists", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Email address is already taken", async () => {
```

File: tests/unit/views/SignUp.spec.ts [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+    const button = wrapper.find("#submit");
+    button.trigger("click");
+
+    await flushPromises();
+    expect(wrapper.find(".v-messages__message").text()).toBe(
+      "Passwords don't match."
+    );
+  });
+
+  it("Submit button is disabled when form validation has a error", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/SignUp.spec.ts]:
```suggestion
  it("Submit button is disabled if form validation has an error", async () => {
```

File: tests/unit/views/home/FollowingActors.spec.ts [hunk 28]
```diff
@@ -25,14 +26,15 @@ describe("FollowingActors.vue", () => {
     store.dispatch("Following/fetchFollowing", "john.doe");
   });
 
-  it("renders all following users as list items", () => {
+  it("renders all following users as list items", async () => {
```
Comment (TiMESPLiNTER) [inline tests/unit/views/home/FollowingActors.spec.ts]: Shall test descriptions start with capital letters or not?

File: tests/unit/views/home/post/text-type/ActivityStreamsArticleType.spec.ts [hunk 29]
```diff
@@ -14,11 +13,11 @@ Vue.use(Vuetify);
 describe("ActivityStreamsArticleType.vue", () => {
   // eslint-disable-next-line
   let vuetify: any;
-  let article: Article;
+  let article: Activity;
```
Comment (TiMESPLiNTER) [inline tests/unit/views/home/post/text-type/ActivityStreamsArticleType.spec.ts]:
```suggestion
  let activity: Activity;
```

File: tests/unit/views/home/post/text-type/ActivityStreamsArticleType.spec.ts [hunk 30]
```diff
@@ -14,11 +13,11 @@ Vue.use(Vuetify);
...
 
   beforeEach(() => {
     vuetify = new Vuetify();
-    const articles = collection.orderedItems as Article[];
+    const articles = collection.orderedItems as Activity[];
```
Comment (TiMESPLiNTER) [inline tests/unit/views/home/post/text-type/ActivityStreamsArticleType.spec.ts]:
```suggestion
    const activities = collection.orderedItems as Activity[];
```

**Comment by fgervasi** [comment]
I merge when the activitypub-objects library is on the master branch

**Pull request merged**